### PR TITLE
[CRO] NikitK/CRO-380/tracking issues for partners signup

### PR DESCRIPTION
**Pull request by NikitK-deriv**
Changes:

In partner signup form we have tracking system for analytic purpose. In that system we have to fix issue with nulls for partners_signup_error_message field and not valid step_codename.

Include default browser API's pop up which handle when user want to close tab/refresh page

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/signup-affiliates/index.tsx [hunk 1]
```diff
@@ -86,7 +90,13 @@ const AffiliateSignup = () => {
...
     useEffect(() => {
         trackEvent({ action: 'open' })
-        return () => trackEvent({ action: 'close' })
+        const handleBeforeUnload = (event) => {
+            event.preventDefault()
+            event.returnValue = 'Are you sure you want to leave this page?'
+            trackEvent({ action: 'close' })
+        }
+        window.addEventListener('beforeunload', handleBeforeUnload)
+        return () => window.removeEventListener('beforeunload', handleBeforeUnload)
```
Comment (prince-deriv) [inline src/pages/signup-affiliates/index.tsx:99]: @NikitK-deriv  won't this get triggered every time we move to another page?

Comment (NikitK-deriv) [reply src/pages/signup-affiliates/index.tsx:99]: Nope, only when user want to close the tab. From this page we can only navigate back to partners page and in that particular case this action does not occur

File: src/pages/signup-affiliates/components/wizard-component/wizard-footer.tsx [hunk 2]
```diff
@@ -58,21 +58,23 @@ const WizardFooter = ({
 }: WizardComponentTypes) => {
     const buttonHandler = React.useCallback(
         (button_type: ButtonType): void => {
+            const prev_step = step - 1
```
Comment (dan-deriv) [inline src/pages/signup-affiliates/components/wizard-component/wizard-footer.tsx:61]: Why do you use different naming notation for variables,  eg. `const buttonHandler`, `const prev_step`?

File: src/pages/signup-affiliates/index.tsx [hunk 3]
```diff
@@ -86,7 +90,13 @@ const AffiliateSignup = () => {
 
     useEffect(() => {
         trackEvent({ action: 'open' })
-        return () => trackEvent({ action: 'close' })
+        const handleBeforeUnload = (event) => {
+            event.preventDefault()
+            event.returnValue = 'Are you sure you want to leave this page?'
```
Comment (dan-deriv) [inline src/pages/signup-affiliates/index.tsx:95]: Shall this be translated?

File: src/pages/signup-affiliates/components/wizard-component/wizard-footer.tsx (hunk 2 above)
Comment (NikitK-deriv) [reply src/pages/signup-affiliates/components/wizard-component/wizard-footer.tsx:61]: @dan-deriv as far as I know in deriv-com we're using snake_case for variables and camelCase for functions/hooks

File: src/pages/signup-affiliates/index.tsx (hunk 3 above)
Comment (NikitK-deriv) [reply src/pages/signup-affiliates/index.tsx:95]: This is banner which occur when user want to close tab
It's my mistake that I asked chatGPT how to handle it and didn't check that value
Screenshot 2023-12-26 at 15 02 05

File: src/pages/signup-affiliates/components/_signup-form.tsx [hunk 4]
```diff
@@ -172,6 +173,8 @@ const AffiliateSignupForm = ({
                     secondary
                     onClick={() => {
                         window.scrollTo(0, 0)
+                        console.log(affiliate_account.email)
```
Comment (mitra-deriv) [inline src/pages/signup-affiliates/components/_signup-form.tsx:176]: Do we need this console log @NikitK-deriv ? 🤔

Comment (habib-deriv) [inline src/pages/signup-affiliates/components/_signup-form.tsx:176]: Please remove the console log
```suggestion
```

**Comment by habib-deriv** [comment]
@NikitK-deriv 
Few smoke test failed. Could you please check if it's related to this PR? Then we'll need a PR to fix the smoke tests as well.

File: src/pages/signup-affiliates/utils/_utils.tsx [hunk 5]
```diff
@@ -69,3 +69,22 @@ export const Submit = ({
             }),
         })
 }
+
+export const trackEvent = ({
+    action,
+    email,
```
Comment (mitra-deriv) [inline src/pages/signup-affiliates/utils/_utils.tsx:75]: does email need to be added to the type and specify type for it ? 

Screenshot 2024-01-03 at 2 59 35 PM

File: src/pages/signup-affiliates/components/_signup-form.tsx (hunk 4 above)
Comment (NikitK-deriv) [reply src/pages/signup-affiliates/components/_signup-form.tsx]: resolved

Comment (NikitK-deriv) [reply src/pages/signup-affiliates/components/_signup-form.tsx]: nope, I already removed it :)

File: src/pages/signup-affiliates/utils/_utils.tsx (hunk 5 above)
Comment (NikitK-deriv) [reply src/pages/signup-affiliates/utils/_utils.tsx:75]: yes @mitra-deriv that should be added to types
I'll do it once `@deriv/analytics` will be updated

**Review by mitra-deriv — DISMISSED** [review]

**Review by aswathy-deriv — APPROVED** [review]

**Pull request merged**